usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -287,6 +278,30 @@ def query_elasticsearch(self) -> list:
         filter_query = QueryWithFilters.generate_awards_elasticsearch_query(self.filters)
         sort_field = self.get_elastic_sort_by_fields()
         sorts = [{field: self.pagination["sort_order"]} for field in sort_field]
+        record_num = (self.pagination["page"] - 1) * self.pagination["limit"]
+        if record_num >= settings.ES_AWARDS_MAX_RESULT_WINDOW and not self.last_id and not self.last_value:
+            sorts.append({"award_id": self.pagination["sort_order"]})
+            logger.warning(
+                "Jumping to random page past 50,000 records will have a performance hit when using Elasticsearch"
+            )
+            self.pagination["upper_bound"] = record_num
+            self.pagination["lower_bound"] = record_num - 1
+            queryset = self.construct_queryset()
+            if len(queryset) != 1:
+                return {}
+            results = {}
+            for result in queryset:
+                results["award_id"] = result.get("award_id")
+                results["total_obligation"] = float(result.get("total_obligation"))
```

Comment:
Since you already forcibly  checked it to have 1 result, maybe no loop and just get the one value, and not results dict?
```
last_record_award_id = queryset[0].get("award_id")
last_record_total_obligation = float(queryset[0].get("total_obligation"))
# ...
.extra(search_after=[last_record_total_obligation, last_record_award_id])[0 : self.pagination["limit"]]
```
?

Reply:
yeah that's fair, will fix